### PR TITLE
monasca: Add support for monasca-log-agent

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

File: chef/cookbooks/monasca/recipes/common.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+# end
+if node["roles"].include?("monasca-log-agent")
+  la_keystone = node[:monasca][:log_agent][:keystone]
+  agents_settings.push(la_keystone)
+end
+
+unless agents_settings.empty?
+  agents_settings.each do |as|
+
+    as_service_user = as["service_user"]
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/common.rb]: nit: these variables are imo not needed. they are not even shorter than ```as["..."]

File: chef/cookbooks/monasca/recipes/common.rb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  user: keystone_settings["admin_user"],
+  password: keystone_settings["admin_password"],
+  tenant: keystone_settings["admin_tenant"]
+}
+
+monasca_pkgs.each do |pkg|
+  package pkg
+end
+
+keystone_register "monasca:common wakeup keystone" do
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/common.rb]: if we want HA we need a sync mark. But can be done later

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    log_agent_dimensions: log_agent_dimensions,
+    keystone_settings: keystone_settings
+  )
+  notifies :reload, "service[monasca-log-agent]"
+end
+
+service "monasca-log-agent" do
+  service_name log_agent_settings[:service_name]
+  supports status: true, restart: true, start: true, stop: true
+  action [:disable, :stop]
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: I might missunderstand something here, but why are we stopping the service?

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  ignore_failure true
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: and why is failing ok here?

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+end
+
+node.save
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: we might want a ```dirty``` flag to not always save the node (but not important for now)

File: chef/cookbooks/monasca/recipes/common.rb [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    role_name role
+    action :add_role
+  end
+end
+
+### recipes agents specific settings
+### single credentials all metrics & logs agents
+agents_settings = []
+
+# once agent is ready, uncomment following lines
```
Comment (jgrassler) [inline chef/cookbooks/monasca/recipes/common.rb]: We should remove this commented code.

File: chef/cookbooks/monasca/recipes/common.rb [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+# if node["roles"].include?("monasca-agent")
+#   agents_settings.push(node[:monasca][:agent][:keystone])
+# end
+if node["roles"].include?("monasca-log-agent")
+  la_keystone = node[:monasca][:log_agent][:keystone]
```
Comment (jgrassler) [inline chef/cookbooks/monasca/recipes/common.rb]: Is this credential information really only available on nodes that have the `monasca-log-agent` role assigned? For if that's not the case we do not need this check.

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+directory "/var/log/monasca-log-agent/" do
+  owner log_agent_settings["user"]
+  group log_agent_settings["group"]
+  mode 0o755
+  recursive true
+end
+directory "/etc/monasca-log-agent/" do
+  owner log_agent_settings["user"]
```
Comment (jgrassler) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: That user/group thing is obsolete: monasca-log-agent is running as root now.

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  group log_agent_settings["group"]
+  mode 0o755
+  recursive true
+end
+
+template "/etc/monasca-log-agent/agent.conf" do
+  source "log-agent.conf.erb"
+  owner log_agent_settings["user"]
```
Comment (jgrassler) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: That user/group thing is obsolete: monasca-log-agent is running as root now.

File: chef/cookbooks/monasca/recipes/common.rb (hunk 1 above)
Comment (dirkmueller) [inline chef/cookbooks/monasca/recipes/common.rb]: agreed, removed

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+log_files = {
+  "/var/log/messages" => "system",
+  "/var/log/zypper.log" => "system"
+}
+
+template "/etc/monasca-log-agent/agent.conf" do
+  source "log-agent.conf.erb"
+  owner log_agent_settings["user"]
```
Comment (jgrassler) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: We no longer have a dedicated user/group for openstack-monasca-log-agent - it's running as root now.

File: chef/data_bags/crowbar/template-monasca.json [hunk 11]
```diff
@@ -2,51 +2,69 @@
...
+        "log_level": "INFO",
+        "keystone": {
+          "service_user": "monasca-metrics-agent",
+          "service_password": "",
+          "service_tenant": "monasca",
+          "service_role": "monasca-agent"
+        }
+      },
+      "log_agent": {
+        "user": "monasca-log-agent",
```
Comment (jgrassler) [inline chef/data_bags/crowbar/template-monasca.json]: We no longer have a dedicated monasca-log-agent user/group. monasca-log-agent is running as root now.

File: chef/data_bags/crowbar/template-monasca.schema [hunk 12]
```diff
@@ -12,34 +12,62 @@
...
+                  }
                 }
-              },
+              }
+            },
+            "log_agent": {
+              "required": true,
+              "type": "map",
+              "mapping": {
+                "user": { "type": "str", required: true },
```
Comment (jgrassler) [inline chef/data_bags/crowbar/template-monasca.schema]: We no longer have a dedicated monasca-log-agent user/group. It's running as root now.

File: chef/cookbooks/monasca/attributes/default.rb [hunk 13]
```diff
@@ -20,7 +21,16 @@
...
+# metric-agent default service settings
+default[:monasca][:metric_agent]["user"] = "monasca-agent"
+default[:monasca][:metric_agent][:group] = "monasca"
+default[:monasca][:metric_agent][:debug] = false
+default[:monasca][:metric_agent]["log_dir"] = "/var/log/monasca-metric-agent"
+default[:monasca][:metric_agent]["agent_service_name"] = "openstack-monasca-metric-agent"
+
+# log-agent default service settings
+default[:monasca][:log_agent][:service_name] = "monasca-log-agent"
+default[:monasca][:log_agent][:user] = "root"
```
Comment (jgrassler) [inline chef/cookbooks/monasca/attributes/default.rb]: We no longer have a dedicated monasca-log-agent user/group. It's running as root now.

**Pull request merged**